src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java

```
@@ -686,4 +687,54 @@ public void register(CacheListener listener) {
 	public void cleanup() {
 		cacheListeners.forEach(CacheListener::cleanup);
 	}
+
+	public boolean isAsyncQueryRunning(String asyncQueryToken) throws SQLException {
+		return getAsyncQueryStatus(asyncQueryToken).equals("RUNNING");
+	}
+
+	public boolean isAsyncQuerySuccessful(String asyncQueryToken) throws SQLException {
+		return getAsyncQueryStatus(asyncQueryToken).equals("ENDED_SUCCESSFULLY");
+	}
+
+	public boolean cancelAsyncQuery(String asyncQueryToken) throws SQLException {
+		if (StringUtils.isBlank(asyncQueryToken)) {
+			throw new FireboltException("Async query token cannot be null or empty");
+		}
+		String asyncQueryId;
+		try (PreparedStatement statement = createPreparedStatement("CALL fb_GetAsyncStatus(?)")) {
+			statement.setString(1, asyncQueryToken);
+			try (ResultSet rs = statement.executeQuery()) {
+				if (!rs.next()) {
+					throw new FireboltException("Could not get query_id for the async query with token: " + asyncQueryToken);
+				}
+				asyncQueryId = rs.getString("query_id");
+			}
+		}
+		if (StringUtils.isNotBlank(asyncQueryId)) {
+			try (PreparedStatement statement = createPreparedStatement("CANCEL QUERY WHERE query_id = ?")) {
+				statement.setString(1, asyncQueryId);
+				statement.executeUpdate();
+			}
```

Do we silently fail here? Let's try to raise the error so the user knows cancelling a query did not succeed.

---

We actually don't silently fail. If the cancel query fails, the error is not being caught anywhere, so it will propagate. there is a test for it in AsyncQueryTest